### PR TITLE
script: add initial version of ts-smoosh

### PR DESCRIPTION
**Pull request by btford**

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]
@btford This looks interesting, if I understand correctly this can help automate a transition from `d.ts` to `.ts`?

If it works well here in kepler we could potentially use this in all our open source frameworks as they currently use `d.ts` files. I could see this feature going into `ocular-dev-tools`. The code looks pretty isolated here so should be easy to move if we decide to do so.

Looking forward to learning about how easy this makes the transition.

@ilijapuaca

**Comment by btford** [comment]
Yes, that is the intent. I've eschewed dependencies (besides TypeScript itself) so this should be fairly portable.

One issue however is that this needs some of the newer compiler transform APIs, which have only been exposed in TypeScript 4.x. This means we either need to update kepler and friends to TS 4 (I'm not sure how big of a lift that would be, but anecdotally TS has been pretty easy to upgrade in other projects), or we need to add just a dependency to TS 4 to this script.

**Comment by ibgreen** [comment]
Sounds good. I've already moved some of the frameworks to ts 4. In most cases it was a 10 minute effort to deal with a couple of new errors.

File: scripts/ts-smoosh/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+# Use
+
+Given a JavaScript file (or list of files) like so:
+
+```
+$ node ts-smoosh/bin ./src/some-file.js
+```
+
+Will produce `.jsx` files using nearby `.d.ts` files.
```
Comment (chrisgervang) [inline scripts/ts-smoosh/README.md:14]: Does it produce .jsx or .tsx?

Comment (heshan0131) [reply scripts/ts-smoosh/README.md:14]: it should produce `.ts`

**Pull request merged**